bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerCreateOp.java

```diff
@@ -200,4 +209,220 @@ private void createComplete(int rc, LedgerHandle lh) {
         cb.createComplete(rc, lh, ctx);
     }
 
+    private static final byte[] EMPTY_PASSWORD = new byte[0];
+
+    public static class CreateBuilderImpl implements CreateBuilder {
+
+        private final BookKeeper bk;
+        private int builderEnsembleSize = 3;
+        private int builderAckQuorumSize = 2;
+        private int builderWriteQuorumSize = 2;
+        private byte[] builderPassword = EMPTY_PASSWORD;
+        private org.apache.bookkeeper.client.api.DigestType builderDigestType
+            = org.apache.bookkeeper.client.api.DigestType.CRC32;
+        private Map<String, byte[]> builderCustomMetadata = Collections.emptyMap();
+        private CreateAdvBuilder advBuilder;
+
+        public CreateBuilderImpl(BookKeeper bk) {
+            this.bk = bk;
+        }
+
+        @Override
+        public CreateBuilder withEnsembleSize(int ensembleSize) {
+            this.builderEnsembleSize = ensembleSize;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withWriteQuorumSize(int writeQuorumSize) {
+            this.builderWriteQuorumSize = writeQuorumSize;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withAckQuorumSize(int ackQuorumSize) {
+            this.builderAckQuorumSize = ackQuorumSize;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withPassword(byte[] password) {
+            this.builderPassword = password;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withCustomMetadata(Map<String, byte[]> customMetadata) {
+            this.builderCustomMetadata = customMetadata;
+            return this;
+        }
+
+        @Override
+        public CreateBuilder withDigestType(org.apache.bookkeeper.client.api.DigestType digestType) {
+            this.builderDigestType = digestType;
+            return this;
+        }
+
+        @Override
+        public CreateAdvBuilder makeAdv() {
```

Comment:
why do we need separate advbuilder? cannt we just assume / implement the logic as if it is advbuilder if ledger id is provided

Reply:
WriteAdvHandle and WriteHandle have different semantics it is better to have a brand new API in which they are completely indipendent.
You can create a LedgerHandleAdv even without providing a ledgerId, if you do not provide it a ledger id is assigned using LedgerIdGenerator